sdk/opa.go

```diff
@@ -0,0 +1,191 @@
+// Copyright 2021 The OPA Authors.  All rights reserved.
+// Use of this source code is governed by an Apache2
+// license that can be found in the LICENSE file.
+
+package sdk
+
+import (
+	"context"
+	"crypto/rand"
+	"fmt"
+	"io/ioutil"
+	"time"
+
+	"github.com/open-policy-agent/opa/bundle"
+	initload "github.com/open-policy-agent/opa/internal/runtime/init"
+	"github.com/open-policy-agent/opa/internal/uuid"
+	"github.com/open-policy-agent/opa/loader"
+	"github.com/open-policy-agent/opa/metrics"
+	"github.com/open-policy-agent/opa/plugins"
+	"github.com/open-policy-agent/opa/plugins/discovery"
+	"github.com/open-policy-agent/opa/plugins/logs"
+	"github.com/open-policy-agent/opa/rego"
+	"github.com/open-policy-agent/opa/server"
+	"github.com/open-policy-agent/opa/storage"
+	"github.com/open-policy-agent/opa/storage/inmem"
+)
+
+// OPA represents an instance of the policy engine.
+type OPA struct {
+	manager     *plugins.Manager
+	configBytes []byte
+	paths       []string
+	filter      loader.Filter
+	bundleMode  bool
+}
+
+// ConfigFile sets the configuration file to use on the OPA instance.
+func ConfigFile(fileName string) func(opa *OPA) error {
```

Comment:
[nit] here and below
```suggestion
func ConfigFile(fileName string) func(*OPA) error {
```

Reply:
OK, will change.. but this feels like something a formatter/linter should do for us if we have an agreed upon convention.

For return values I've started to appreciate named ones just recently as I discovered IntelliJ uses them nicely when generating code and docs.